File: api/src/main/java/net/impactdev/gts/api/events/extension/ExtensionLoadEvent.java
package net.impactdev.gts.api.events.extension;

import net.impactdev.impactor.api.event.ImpactorEvent;
import net.impactdev.gts.api.extension.Extension;
import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;

@GenerateFactoryMethod
public interface ExtensionLoadEvent extends ImpactorEvent {

    Extension getExtension();

}

File: api/src/main/kotlin/net/impactdev/gts/api/events/extension/ExtensionLoadEvent.kt
package net.impactdev.gts.api.events.extension

import net.impactdev.gts.api.extension.Extension
import net.impactdev.impactor.api.event.ImpactorEvent

interface ExtensionLoadEvent : ImpactorEvent {
    @get:Param(0)
    val extension: Extension
}